src/monitorAnimationSpeed.js

```diff
@@ -0,0 +1,43 @@
+import React, {Component} from 'react'
+import collectFps from 'collect-fps'
+
+const noop = () => {}
+
+export default ({
+  threshold = 30,
+  fpsCollector = collectFps
+}) => (Target) => {
+  let endFPSCollection = noop
+
+  const handleStartFPSCollection = (component) => {
+    if (endFPSCollection !== noop) { endFPSCollection() }
+
+    endFPSCollection = fpsCollector()
+  }
+
+  class MonitorAnimationSpeed extends Component {
+    constructor () {
+      super()
+
+      this.state = {fps: 60}
```

Comment:
I think this should start at `threshold`, so that there's no chance that it will trigger `lowFPS`, regardless of how low `threshold` is.